packages/stimulus/src/index.ts

```diff
@@ -3,9 +3,16 @@ import type { JSClient, JSSpan } from "@appsignal/types"
 export function installErrorHandler(appsignal: JSClient, application: any) {
   const prevHandler = application.handleError
 
-  application.handleError = function (error: Error, message: string) {
+  application.handleError = function (
+    error: Error,
+    message: string,
+    detail: { identifier: string }
+  ) {
     const span = appsignal.createSpan((span: JSSpan) =>
-      span.setTags({ framework: "Stimulus", message }).setError(error)
+      span
+        .setAction(`${detail.identifier}-controller`)
```

Comment:
> but my guess is that it is trying to make the action "look like" [the file location that this identifier is derived from](https://stimulus.hotwired.dev/reference/controllers#identifiers)?

That was exactly my intention.

> Do you think it be useful to attempt to use detail.controller.constructor.name as the action name?

This makes a lot of sense, but based on common usage, I think we should expect it to be an anonymous class.

It would be possible to use the class name and fall back to the identifier with something like...

```js
let name = detail.controller.constructor.name
if (name === "extended") { name = detail.identifier }
```

I suggest keeping it simpler, but let me know if you prefer this.

---

As a proposal, I'm going to change this to just use the identifier. Although it may be less obvious what it refers to, stimulus developers will be comfortable with it from defining `data-controller="identifier"`. This feels like the safest and least-surprising approach.

That said, I'm happy to continue discussing the most appropriate default for this.

Reply:
> I think we should expect it to be an anonymous class. It would be possible to use the class name and fall back to the identifier [...] I suggest keeping it simpler

Agreed. I also think it would be confusing if some controllers ended up showing the class name while others used the identifier.

> As a proposal, I'm going to change this to just use the identifier. Although it may be less obvious what it refers to, stimulus developers will be comfortable with it from defining data-controller="identifier". This feels like the safest and least-surprising approach.

That sounds reasonable to me as well! Go ahead 💯👍

Reply:
ok, I think I've addressed this and your other comment in 1a94c196cb33274238a221f130a324aadc5c55fa.